src/mcp_agent/mcp/mcp_aggregator.py

```
@@ -374,183 +453,343 @@
+                    span.set_attribute(
+                        f"{server_name}.capabilities.{attr}", value is not None
+                    )
+
+            if self.connection_persistence:
                 try:
-                    initialize_result = await session.initialize()
-                    return initialize_result.capabilities
+                    server_conn = await self._persistent_connection_manager.get_server(
+                        server_name, client_session_factory=MCPAgentClientSession
+                    )
+                    # TODO: saqadri (FA1) - verify
```

Pylint (3.3.7)

[warning] TODO: saqadri (FA1) - verify

(W0511)